File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums){
            set.add(num);
        }

        int longestSequenceLength = 0;

        for(int num : nums){
            int currSequenceLength = 1;
            longestSequenceLength = Math.max(longestSequenceLength, currSequenceLength);
            if(set.contains(num-1)){
                continue;
            }
            while(set.contains(num+1)){
                currSequenceLength++;
                longestSequenceLength = Math.max(longestSequenceLength, currSequenceLength);
                num++;
            }
        }
        return longestSequenceLength;
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        
        if n == 1:
            return nums[0]
        elif n == 2:
            return max(nums)
        elif n == 3:
            return max(nums[0] + nums[2], nums[1])
        
        result = [0] * len(nums)
        result[0] = nums[0]
        result[1] = nums[1]
        result[2] = nums[0] + nums[2]

        for i in range(3, n):
            result[i] = nums[i] + max(result[i-3], result[i - 2])
        
        return max(result[n - 2], result[n -1])
    